fossilize_db.cpp

```diff
@@ -119,6 +119,9 @@ struct ExportedMetadataHeader
 };
 static_assert(sizeof(ExportedMetadataHeader) % 8 == 0, "Alignment of ExportedMetadataHeader must be 8.");
 
+// Allow termination request if using the interface on a thread
+std::atomic<bool> shutdown_requested;
```

Comment:
Is there a need for this to be a static global? I would imagine something like `database_iface->request_shutdown()` would work too.

Reply:
That was my first implementation, but when the extra read only database was used it wouldn't terminate those.  I would have to connect those somehow to the other database I guess?

Reply:
Right ye ... Could get spicy w.r.t. thread safety as well.